utils/docker/vcluster/README.md

```diff
@@ -0,0 +1,199 @@
+
+To build and deploy the Docker images, `docker` and optionally `docker-compose` shall be available.
+The docker host should have access to the [Docker Hub](https://hub.docker.com/) and
+[CentOS](https://www.centos.org/) official repositories.  Finally,
+[hugepages](https://www.kernel.org/doc/Documentation/vm/hugetlbpage.txt) linux kernel feature shall
+be enabled on the docker host.
+
+The platform was tested and validated with the [CentOS8](https://hub.docker.com/_/centos) official
+docker image.  However other RHEL-like distributions such as
+[rocky8.4](https://hub.docker.com/r/rockylinux/rockylinux) should be supported.
+
+!!! warning
+    Some distributions are not yet well supported such as
+    [rocky8.5](https://hub.docker.com/r/rockylinux/rockylinux): issue with the management of
+    hugepages with the [spdk](https://spdk.io/) library.
+
+
+## Building Docker Images
+
+### Building Base DAOS Image
+
+The first image to create is the `daos-base` image which is not intetended to be used as it, but as
+a base image for building the other three daos images.  This first image could be built directly
+from GitHub with the following command:
+
+```bash
+$ docker build --tag daos-base:centos8 \
+	https://github.com/daos-stack/daos.git#master:utils/docker/vcluster/daos-base/el8
+```
+
+or from a local tree with the following command:
+
+```bash
+$ docker build --tag daos-base:centos8 utils/docker/vcluster/daos-base/el8
+```
+
+This Docker file accept the following arguments:
+
+- `RHEL_BASE_IMAGE`: Base docker image to use (default centos)
+- `RHEL_BASE_VERSION`: Version of the base docker image to use (default 8)
+- `BUST_CACHE`: Manage docker building cache (default undefined).  To invalidate the cache, a random
+	value such as the date of the day shall be given.
+- `DAOS_AUTH`: Enable DAOS authentication when set to "yes" (default "no")
+- `DAOS_REPOS`: Space separated list of repos needed to install DAOS (default
+	"https://packages.daos.io/v2.0")
+- `DAOS_GPG_KEYS`: Space separated list of GPG keys associated with DAOS repos (default
+	"https://packages.daos.io/RPM-GPG-KEY")
+- `DAOS_REPOS_NOAUTH`: Space separated list of repos to use without GPG authentication
+	(default "")
+
+For example, building a DAOS base image with authentication enabled could be done with the
+following command:
+
+```bash
+$ docker build --tag daos-base:centos8 --build-arg DAOS_AUTH=yes \
+	https://github.com/daos-stack/daos.git#master:utils/docker/vcluster/daos-base/el8
+```
+
+### Building DAOS Nodes Images
+
+The three images `daos-server`, `daos-admin` and `daos-client` could be built directly from GitHub
+or from a local tree in the same way as for the `daos-base` image.  Following command could be used
+to build directly the three images from GitHub:
+
+```bash
+$ for image in daos-server daos-admin daos-client ; do \
+	docker build --tag "$image:centos8" \
+		"https://github.com/daos-stack/daos.git#master:utils/docker/vcluster/$image/el8"; \
+  done
+```
```

Comment:
I wondered if this step was necessary, but it may be if there isn't a way for the docker-compose command to be run from github, I found it slightly confusing to have an optional step in the middle of the instructions.

Reply:
Indeed, docker-compose does not yet support github as input :'(
From what I understand of the US, it was required that all step should be able to be run without having access to the source tree.